classes/chat/ChatManager.php

```diff
@@ -78,33 +78,75 @@ private function fetchPosts(?int $starting_post_id = null, int $max_posts = self
         $query .= " ORDER BY `post_id` DESC LIMIT $max_posts";
 
         $result = $this->system->db->query($query);
-
+        $result = $this->system->db->fetch_all($result);
+
+        // Get user IDs from posts
+        $user_ids = [];
+        $user_data_by_id = [];
+        $user_data_by_name = [];
+        foreach ($result as $row) {
+            // If user_id is set (newer post) add to list for batch
+            if ($row['user_id'] > 0 && !in_array($row['user_id'], $user_ids)) {
+                $user_ids[] = $row['user_id'];
+            }
+            // If user_id is not set (older post) get data for that user based on user_name
+            else {
+                $user_data_result = $this->system->db->query("
+                    SELECT `users`.`user_id`, `users`.`staff_level`, `users`.`premium_credits_purchased`, `users`.`chat_effect`, `users`.`avatar_link`,
+                    `user_settings`.`avatar_style`, `user_settings`.`avatar_frame`
+                    FROM `users`
+                    LEFT JOIN `user_settings` ON `users`.`user_id` = `user_settings`.`user_id`
+                    WHERE `users`.`user_name` = '{$this->system->db->clean($row['user_name'])}'
+                ");
+                $user_data_result = $this->system->db->fetch($user_data_result);
+                // only add data to array if user match found
+                if ($this->system->db->last_num_rows > 0) {
+                    // if custom setting is not set
+                    if (!isset($user_data_result['avatar_style'])) {
+                        $user_data_result['avatar_style'] = "avy_round";
+                    }
+                    // if custom setting is not set
+                    if (!isset($user_data_result['avatar_frame'])) {
+                        $user_data_result['avatar_frame'] = "avy_frame_default";
+                    }
+                    $user_data_by_name[$row['user_name']] = $user_data_result;
+                }
+            }
+        }
```

Comment:
Doesn't appear that this handles the situation where there is no UID and user isn't found by name.

Reply:
There was a check for if user data is found for a post since this case can currently happens as well (when users change their name). Updated it slightly here to check if data was found either by ID or name:
```            //Fetch user data
            $user_data = null;
            if (isset($user_data_by_id[$post->user_id])) {
                $user_data = $user_data_by_id[$post->user_id];
            }
            else if (isset($user_data_by_name[$post->user_name])) {
                $user_data = $user_data_by_name[$post->user_name];
            }```